src/probnum/diffeq/perturbedsolvers/_perturbation_functions.py

```diff
@@ -0,0 +1,50 @@
+"""perturbation functions to perturb the stepsize."""
+import numpy as np
+
+
+def perturb_uniform(step, order, noise_scale, seed=1):
+    """perturbs the step with uniformly distributed noise scaled by noise-scale.
+    proposed by Abdulle and Garegnani(2020)
+
+    Parameters
+    ----------
+    step : float
+        unperturbed step propesed by the steprule
+    order : int
+        order of the solver
+    noise_scale : float
+        scales the perturbation
+    seed : int
+        seed for pseudo-random number generator.
+    """
+    np.random.seed(seed=seed)
+    if step < 1:
+        noisy_step = np.random.uniform(
```

Comment:
see my random state and sampling comment above. This is a more flexible way than setting np.random.seed to a fixed value.